### PR TITLE
Issue/5877 new inbox screen

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Wip for: #5877

### Description
Displays a list of notes inside the new Inbox screen. The data is hard-coded at the moment. Just working on the UI side. 
This feature is hidden under a feature flag so it is safe to merge

### Testing instructions
Open Inbox and check design matches designs vbzgTKHV9XsyyN5cxxAuhv-fi-2669%3A91520

### Images/gif

| Dark  | Light |
| ------------- | ------------- |
|   |   |
| Content Cell  | Content Cell  |

- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/inbox/InboxFragment.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentInboxBinding.inflate(inflater, container, false)
+
+        val view = binding.root
+        binding.inboxComposeView.apply {
+            // Dispose of the Composition when the view's LifecycleOwner is destroyed
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MdcTheme {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/inbox/InboxFragment.kt:34]: We don't use MaterialTheme anymore. With MdcTheme we can then access some of the existing attributes defined in our `themes.xml`. 

More on this [here](https://material-components.github.io/material-components-android-compose-theme-adapter/)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/inbox/InboxViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    )
+
+    data class InboxNoteUi(
+        val id: String,
+        val title: String,
+        val description: String,
+        val updatedTime: String,
+        val callToActionText: String,
+        val onCallToActionClick: (String) -> Unit,
+        val dismissText: String,
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/inbox/InboxViewModel.kt:91]: ❓ Are there anticipated cases where the dismiss text needs to be different between one note to another?

**Review by hafizrahman — COMMENTED** [review]
This looks good, thanks for working on it! I left a question and added Preview.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/inbox/InboxViewModel.kt (hunk 2 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/inbox/InboxViewModel.kt:91]: Good point. Not that I know of. But still, we need a field inside the UI model to store the string. We want to provide all the needed information to the compose function and avoid having to retrieve any context.getString() from the view/compose code.

**Comment by JorgeMucientes** [comment]
Thanks for reviewing this @hafizrahman and for adding the missing preview :)

**Review by hafizrahman — APPROVED** [review]
LGTM, thanks and sorry for the delay due to my AFKs 🤦🏼

**Pull request merged**